store/sqlstore/context.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import "context"
+
+// storeContextKey is the base type for all context keys for the store.
+type storeContextKey string
+
+// contextValue is a type to hold some pre-determined context values.
+type contextValue string
+
+// Different possible values of contextValue.
+const (
+	useMaster contextValue = "useMaster"
+)
+
+// withMaster adds the context value that master DB should be selected for this request.
+func withMaster(ctx context.Context) context.Context {
+	return context.WithValue(ctx, storeContextKey(useMaster), "yes")
```

Comment:
`"yes"` vs `true`, what's the reason for choosing a string?

Reply:
The exact value of yes/true doesn't really matter. We can choose any one. The reason to use string is that we can store more context values in the future, and keep the context key type same. A bool will not allow you to do that. Then you need different keys. It's cleaner to have a single key and different values.

Reply:
Sure, I guess there could be some future values of "useMaster" other than "yes"/"no".

Reply:
Sorry for the confusion here. That's not quite what I mean. To expand further, a context.WithValue needs a "key" and a "value". And the key needs to be of a named type.

Right now, we have - storeContextKey("master"), "yes/no".

If we start to use a boolean, then the value part doesn't have any significance, and we need to have 2 separate keys.

```
useMaster = true
dontUse = false

storeContextKey(useMaster), ???
storeContextKey(dontUse), ???
```

It becomes something  like this: https://play.golang.org/p/a-s4l3Kuwe0

This is still okay, but now we have painted ourselves to a corner. Because `storeContextKey` is of type bool. So we cannot have any more values rather than true/false. Which means we need more context key types. Hence, the need to have string keys.

Does that help or was that even more confusing :P ?

Reply:
I got a bit confused as well :sweat_smile: The ask is around using a boolean in the value rather than the key, right? Something like:

```go
context.WithValue(ctx, storeContextKey(useMaster), true)
```

With this the key type would remain the same and the value would express the "yes"/"no" as well without the need of the `dontUse`. In case we ever need to store another context value, we could reuse the key type but with a different value type. But I feel there is something I'm not seeing here

Reply:
Oops, you are right. My bad. I somehow thought we need to change the key type also. Let me correct that.

Reply:
@mgdelacroix - updated

Reply:
Awesome, thanks!!